tests/unit/rl_worker/test_worker_util.py

```diff
@@ -0,0 +1,174 @@
+        return "/api/challenges/challenge/queues/{}/".format(queue_name)
+
+    def get_challenge_phase_by_pk_url(self, challenge_pk, challenge_phase_pk):
+        return "/api/challenges/challenge/{}/challenge_phase/{}".format(challenge_pk, challenge_phase_pk)
+
+    def update_submission_data_url(self, challenge_pk):
+        return "/api/jobs/challenge/{}/update_submission/".format(challenge_pk)
+
+    def update_submission_status_url(self, challenge_pk):
+        return "/api/jobs/challenge/{}/update_submission/".format(challenge_pk)
+
+
+class WorkerUtilTestClass(BaseTestClass):
+    def setUp(self):
+        super(WorkerUtilTestClass, self).setUp()
+        self.success_response = {"success": "success_message", "status_code": 200}
+        self.example_error_description = "Example description"
+
+    def test_return_url_per_environment(self):
+        returned_url = interface.return_url_per_environment(self.make_request_url())
+        expected = "{0}{1}".format(evalai_api_server, self.make_request_url())
+        self.assertEqual(returned_url, expected)
+
+    @responses.activate
+    def test_make_request_get_success(self):
+        url = "{0}{1}".format(evalai_api_server, self.make_request_url())
+        responses.add(responses.GET, url,
+                      json=self.success_response,
+                      content_type='application/json',
+                      status=200)
+
+        response = interface.make_request(url, "GET")
+
+        self.assertEqual(response, self.success_response)
+
+    @responses.activate
+    @mock.patch("scripts.workers.worker_util.logger.info")
+    def test_make_request_http_error(self, mock_logger):
+        url = "{0}{1}".format(evalai_api_server, self.make_request_url())
+        responses.add(responses.GET, url,
+                      json={"error": "404 Not Found"},
+                      status=404)
+
+        with self.assertRaises(requests.exceptions.HTTPError):
+            interface.make_request(url, "GET")
+
+        mock_logger.assert_called_with("The worker is not able to establish connection with EvalAI")
+
+    @responses.activate
+    @mock.patch("scripts.workers.worker_util.logger.info")
+    def test_make_request_connection_error(self, mock_logger):
+        url = "{0}{1}".format(evalai_api_server, self.make_request_url())
+        responses.add(responses.GET, url,
+                      body=requests.exceptions.RequestException(self.example_error_description))
+
+        with self.assertRaises(requests.exceptions.RequestException):
+            interface.make_request(url, "GET")
+
+        mock_logger.assert_called_with("The worker is not able to establish connection with EvalAI")
+
+    def test_get_message_from_sqs_queue(self):
+        url = "{}{}".format(evalai_api_server, self.get_message_from_sqs_queue_url(queue_name))
+        with mock.patch("__main__.EvalAI_Interface.make_request", return_value=self.success_response) as mock_make_request:
```

Comment:
@vkartik97 @RishabhJain2018 I am currently stuck on this issue since a few days.
The tests for `make_request` are working correctly, so I wanted to mock make_request in other tests, but this always throws an error. I tried using several different ways to patch like `patch("__main__.interface.make_request")` and simply `patch(interface.make_request)` which should work according to docs but the travis throws a `ModuleNotFoundError` with the message `__main__ is not a  package`.  
Is there a workaround for this or should we go with using the `responses` library to write tests?